cmd/activity-monitor/main.go

```diff
@@ -120,11 +96,11 @@ func startMonitor(rpcCh *amqp.Channel, logger *blog.AuditLogger, stats statsd.St
 		cmd.FailOnError(err, "Could not subscribe to queue")
 	}
 
-	deliveryTimings := make(map[string]time.Time)
+	rpcMonitor := metrics.NewRPCMonitor(stats)
 
 	// Run forever.
 	for d := range deliveries {
-		go timeDelivery(d, stats, deliveryTimings)
+		go rpcMonitor.TimeDelivery(d)
```

Comment:
I'm not sure we can do something like that without making major changes to the server/client setup, I've benchmarked this method and it's quite fast, is your problem just the cleanup locking shenanigans?